src/the-parser.md

```diff
@@ -1,68 +1,74 @@
+  1. _Lexing_ takes strings and turns them into streams of [tokens]. For
+  example, `foo.bar + buz` would be turned into the tokens `foo`, `.`, `bar`,
+  `+`, and `buz`.
 
 [tokens]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast/token/index.html
 [lexer]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lexer/index.html
 
-Parsing then takes streams of tokens and turns them into a structured
-form which is easier for the compiler to work with, usually called an [*Abstract
-Syntax Tree*][ast] (AST). An AST mirrors the structure of a Rust program in memory,
-using a `Span` to link a particular AST node back to its source text.
+  2. _Parsing_ takes streams of tokens and turns them into a structured form
+  which is easier for the compiler to work with, usually called an [*Abstract
+  Syntax Tree* (`AST`)][ast] . 
+  
+  
+An `AST` mirrors the structure of a Rust program in memory, using a `Span` to
+link a particular `AST` node back to its source text. The `AST` is defined in
+[`rustc_ast`][rustc_ast], along with some definitions for tokens and token
+streams, data structures/`trait`s for mutating `AST`s, and shared definitions for
+other `AST`-related parts of the compiler (like the lexer and
+`macro`-expansion).
 
-The AST is defined in [`rustc_ast`][rustc_ast], along with some definitions for
-tokens and token streams, data structures/traits for mutating ASTs, and shared
-definitions for other AST-related parts of the compiler (like the lexer and
-macro-expansion).
+The lexer is developed in [`rustc_lexer`][lexer].
 
 The parser is defined in [`rustc_parse`][rustc_parse], along with a
 high-level interface to the lexer and some validation routines that run after
-macro expansion. In particular, the [`rustc_parse::parser`][parser] contains
+`macro` expansion. In particular, the [`rustc_parse::parser`][parser] contains
 the parser implementation.
 
-The main entrypoint to the parser is via the various `parse_*` functions and others in the
-[parser crate][parser_lib]. They let you do things like turn a [`SourceFile`][sourcefile]
+The main entrypoint to the parser is via the various `parse_*` functions and others in
+[rustc_parse][rustc_parse]. They let you do things like turn a [`SourceFile`][sourcefile]
 (e.g. the source in a single file) into a token stream, create a parser from
-the token stream, and then execute the parser to get a `Crate` (the root AST
+the token stream, and then execute the parser to get a [`Crate`] (the root `AST`
 node).
 
-To minimize the amount of copying that is done,
-both [`StringReader`] and [`Parser`] have lifetimes which bind them to the parent `ParseSess`.
-This contains all the information needed while parsing,
-as well as the [`SourceMap`] itself.
+To minimize the amount of copying that is done, both [`StringReader`] and
+[`Parser`] have lifetimes which bind them to the parent [`ParseSess`]. This
+contains all the information needed while parsing, as well as the [`SourceMap`]
+itself.
```

Comment:
please don't convert nice semantic line breaks into hard line breaks like this, it makes it harder to read and harder to diff